### PR TITLE
disable device id logging for demo users

### PR DESCRIPTION
**Pull request by sravfeyn**

### CODE REVIEW

File: corehq/apps/mobile_auth/views.py [hunk 1]
```diff
@@ -66,8 +66,11 @@ def fetch_key_records(request, domain):
     payload = FetchKeyRecords(domain, user_id, last_issued).get_payload()
     device_id = request.GET.get('device_id')
     if device_id and isinstance(request.couch_user, CommCareUser):
-        request.couch_user.update_device_id_last_used(device_id)
-        request.couch_user.save()
+        if not request.couch_user.is_demo_user:
```
Comment (snopoke) [inline corehq/apps/mobile_auth/views.py]: nit, could make this part of the previous `if` condition

File: corehq/apps/mobile_auth/views.py [hunk 2]
```diff
@@ -66,8 +66,11 @@ def fetch_key_records(request, domain):
...
+            # temporary NIC fix; db conflicts happen when multiple logins happen
+            # because of following, so disable it for demo users
```
Comment (snopoke) [inline corehq/apps/mobile_auth/views.py]: Don't think this should be temporary - it doesn't make sense to keep track of device ID's for demo users.

File: corehq/apps/mobile_auth/views.py (hunk 1 above)
Comment (snopoke) [inline corehq/apps/mobile_auth/views.py]: Ah, I see, `is_demo_user` is only an attribute on `CommCareUser`

File: corehq/apps/mobile_auth/views.py (hunk 2 above)
Comment (sravfeyn) [inline corehq/apps/mobile_auth/views.py]: Cool, will update it. (not updating right now, assuming that will restart the build again)

Comment (snopoke) [inline corehq/apps/mobile_auth/views.py]: @sravfeyn fine to merge this once you've updated the comment

**Pull request merged**